crates/circuit/src/circuit_data.rs

```diff
@@ -327,6 +317,25 @@ impl CircuitData {
         Ok(res)
     }
 
+    /// Performs an empty-like shallow copy.
+    ///
+    /// Returns:
+    ///     CircuitData: The shallow copy.
+    #[pyo3(signature = (reserve = None,))]
+    pub fn copy_empty_like(&self, py: Python<'_>, reserve: Option<usize>) -> PyResult<Self> {
+        let mut res = CircuitData::new(
+            py,
+            Some(self.qubits.cached().bind(py)),
+            Some(self.clbits.cached().bind(py)),
+            None,
+            reserve.unwrap_or_default(),
+            self.global_phase.clone(),
+        )?;
+        res.qargs_interner = self.qargs_interner.clone();
+        res.cargs_interner = self.cargs_interner.clone();
```

Comment:
It's not clear to me that this is a good idea for a general `copy_empty_like` function.  It makes sense for `copy` to do it, but in general, this function isn't going to know what instructions are about to be pushed to it, so cloning the whole interner may well be quite wasteful.  Perhaps better to make that a separate method - I think the functionality is already public via the interner-merge stuff.

Reply:
Addressed in dc555c8